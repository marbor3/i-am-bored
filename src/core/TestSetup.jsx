import React from "react";
import { render } from "@testing-library/react";
import { StateProvider } from "../state/DashboardState";
import dashboardReducer, {
  createDashboardIniitalState,
} from "../state/DashboardReducer";

const mockGeolocation = {
  getCurrentPosition: jest.fn().mockImplementation((success) =>
    Promise.resolve(
      success({
        coords: {
          latitude: 51.14,
          longitude: 17.04,
        },
      })
    )
  ),
  watchPosition: jest.fn(),
};

global.navigator.geolocation = mockGeolocation;

const renderWithContext = (component, options) =>
  render(
    <StateProvider
      reducer={dashboardReducer}
      initialState={createDashboardIniitalState(options)}
    >
      {component}
    </StateProvider>
  );

export { renderWithContext };
