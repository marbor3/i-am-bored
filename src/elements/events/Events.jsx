import react, { useEffect, useState } from 'react';
import Event from '../event/Event';
import styles from './Events.module.css';

export default function Events({ events }) {
  return events && events.size ? (
    <ul className={styles.events}>
      {[...events.values()].map((event) =>
        event.isVisible ? (
          <li data-testid="event-list-item" key={event.id} className={styles['events__list-item']}>
            <Event {...event} />
          </li>
        ) : null
      )}
    </ul>
  ) : null;
}
