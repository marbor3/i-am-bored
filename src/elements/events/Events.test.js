import React from "react";
import { render } from "@testing-library/react";
import Events from "./Events";
import { renderWithContext } from "../../core/TestSetup";

describe("Elements - Events", () => {
  test("without any events theres nothing to render", async () => {
    const { queryByTestId, container } = renderWithContext(<Events />);

    expect(queryByTestId("event-list-item")).not.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("with empty events list theres nothing to render", async () => {
    const { queryByTestId, container } = renderWithContext(
      <Events events={[]} />
    );

    expect(queryByTestId("event-list-item")).not.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("given an array of events, it renders all events", async () => {
    const events = new Map();

    events.set("0ga2EdN7prc", {
      id: "0ga2EdN7prc",
      name: "Steve Hackett Genesis Revisited - Seconds Out & More",
      isVisible: true,
    });
    events.set("0oO71TSv4Ed", {
      id: "0oO71TSv4Ed",
      name: "Muzyka zespo≈Çu METALLICA symfonicznie",
      isVisible: true,
    });
    events.set("0oz51ozk3ob", {
      id: "0oz51ozk3ob",
      name: "Happysad",
      isVisible: true,
    });

    const { getAllByTestId, container } = renderWithContext(
      <Events events={events} />
    );

    expect(getAllByTestId("event-list-item")).toHaveLength(3);
    expect(container).toMatchSnapshot();
  });
});
