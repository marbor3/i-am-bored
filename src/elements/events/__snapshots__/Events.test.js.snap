// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Elements - Events given an array of events, it renders all events 1`] = `
<div>
  <ul
    class="events"
  >
    <li
      class="events__list-item"
      data-testid="event-list-item"
    >
      <article
        class="article null"
        data-testid="event"
      >
        <h2>
          <button>
            fav
          </button>
          Steve Hackett Genesis Revisited - Seconds Out & More
        </h2>
        <main />
        <a
          href="/event/0ga2EdN7prc"
        >
          more info
        </a>
      </article>
    </li>
    <li
      class="events__list-item"
      data-testid="event-list-item"
    >
      <article
        class="article null"
        data-testid="event"
      >
        <h2>
          <button>
            fav
          </button>
          Muzyka zespo≈Çu METALLICA symfonicznie
        </h2>
        <main />
        <a
          href="/event/0oO71TSv4Ed"
        >
          more info
        </a>
      </article>
    </li>
    <li
      class="events__list-item"
      data-testid="event-list-item"
    >
      <article
        class="article null"
        data-testid="event"
      >
        <h2>
          <button>
            fav
          </button>
          Happysad
        </h2>
        <main />
        <a
          href="/event/0oz51ozk3ob"
        >
          more info
        </a>
      </article>
    </li>
  </ul>
</div>
`;

exports[`Elements - Events with empty events list theres nothing to render 1`] = `<div />`;

exports[`Elements - Events without any events theres nothing to render 1`] = `<div />`;
